src/webots/gui/WbMainWindow.cpp

```diff
@@ -1615,6 +1629,138 @@ void WbMainWindow::exportHtml() {
 
   WbSimulationState::instance()->setMode(currentMode);
 }
+void WbMainWindow::ShareMenu() {
+  WbSimulationState::Mode currentMode = WbSimulationState::instance()->mode();
+  WbShareWindow *shareWindowUi = new WbShareWindow(this);
+  shareWindowUi->exec();
+  WbSimulationState::instance()->setMode(currentMode);
+}
+
+void WbMainWindow::uploadScene() {
+  WbWorld *world = WbWorld::instance();
+  world->exportAsHtml(WbStandardPaths::webotsTmpPath() + "export_cloud.html", false);
+  sendCloudRequest();
+}
+
+void WbMainWindow::sendCloudRequest() {
+  QNetworkRequest request(QUrl("https://beta.webots.cloud/ajax/animation/create.php"));
```

Comment:
This URL shouldn't be hard-coded this way. Instead, the URL of webots.cloud should be defined in the preferences of Webots, under the "Network" tab and should take `https://beta.webots.cloud` as a default value (to which we will append the `/ajax/animation/create.php` URI).
```suggestion
  QNetworkRequest request(QUrl("https://beta.webots.cloud/ajax/animation/create.php"));
```